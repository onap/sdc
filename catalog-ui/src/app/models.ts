/*-
 * ============LICENSE_START=======================================================
 * SDC
 * ================================================================================
 * Copyright (C) 2017 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END=========================================================
 */

/**
 * Created by ob0695 on 2/23/2017.
 */
import from = require("core-js/fn/array/from");
export * from './models/activity';
export * from './models/additional-information';
export * from './models/app-config';
export * from './models/validation-config';
export * from './models/plugins-config';
export * from './models/artifacts';
export * from './models/aschema-property';
export * from './models/schema-attribute';
export * from './models/attributes';
export * from './models/capability';
export * from './models/category';
export * from './models/comments';
export * from './models/components/component';
export * from './models/components/displayComponent';
export * from './models/components/resource';
export * from './models/components/service';
export * from './models/componentsInstances/componentInstance';
export * from './models/componentsInstances/resourceInstance';
export * from './models/componentsInstances/serviceInstance';
export * from './models/componentsInstances/serviceProxyInstance';
export * from './models/componentsInstances/serviceSubstitutionInstance';
export * from './models/graph/zones/group-instance';
export * from './models/graph/zones/policy-instance';
export * from './models/graph/zones/zone';
export * from './models/graph/zones/zone-instance';
export * from './models/csar-component';
//export * from './models/data-type-properties';
export * from './models/properties-inputs/property-be-model';
export * from './models/properties-inputs/property-fe-model';
export * from './models/properties-inputs/property-fe-map';
export * from './models/properties-inputs/derived-fe-property';
export * from './models/properties-inputs/property-declare-api-model';
export * from './models/properties-inputs/property-input-detail';
export * from './models/properties-inputs/input-fe-model';
export * from './models/properties-inputs/simple-flat-property';
export * from './models/data-types-map';
export * from './models/data-types';
export * from './models/distribution';
export * from './models/export-excel';
export * from './models/file-download';
export * from './models/graph/graph-links/common-base-link';
export * from './models/graph/graph-links/common-ci-link-base';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-link-base';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-simple-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-ucpe-host-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-ucpe-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-vl-link';
export * from './models/graph/graph-links/composition-graph-links/composition-ci-vl-ucpe-link';
export * from './models/graph/graph-links/links-factory';
export * from './models/graph/graph-links/module-graph-links/module-ci-link-base';
export * from './models/graph/graph-links/module-graph-links/module-ci-vl-link';
export * from './models/graph/graphTooltip';
export * from './models/graph/assetPopoverObj';
export * from './models/graph/link-menu';
export * from './models/graph/match-relation';
export * from './models/graph/nodes/base-common-node';
export * from './models/graph/nodes/common-ci-node-base';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-base';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-cp';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-configuration';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service-proxy';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-service-substitution';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-ucpe-cp';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-ucpe';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vf';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vfc';
export * from './models/graph/nodes/composition-graph-nodes/composition-ci-node-vl';
export * from './models/graph/nodes/modules-graph-nodes/module-node-base';
export * from './models/graph/nodes/nodes-factory';
export * from './models/graph/point';
export * from './models/graph/connectRelationModel';
export * from './models/graph/relationship';
export * from './models/heat-parameters';
export * from './models/input-property-base';
export * from './models/inputs-and-properties';
export * from './models/inputs';
export * from './models/instance-inputs-properties-map';
export * from './models/left-panel';
export * from './models/member';
export * from './models/modules/base-module';
export * from './models/operation';
export * from './models/properties';
export * from './models/requirement';
export * from './models/server-error-response';
export * from './models/tab';
export * from './models/tooltip-data';
export * from './models/user';
export * from './models/validate';
export * from './models/component-metadata';
export * from './models/components-array';
export * from './models/modal';
export * from './models/button';
export * from './models/wizard-step';
export * from './models/radio-button';
export * from './models/filter-properties-assignment-data';
export * from './models/properties-inputs/input-be-model';
export * from './models/catalogSelector';
export * from './models/componentsInstances/fullComponentInstance';
export * from './models/catalogSelector';
export * from './models/componentsInstances/fullComponentInstance';
export * from './models/service-instance-properties-and-interfaces';
export * from './models/relationship-types';
export * from './models/tosca-presentation';
export * from './models/node-types';
export * from './models/capability-types';