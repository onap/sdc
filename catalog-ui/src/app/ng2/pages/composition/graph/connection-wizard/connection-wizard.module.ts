import {ToNodeStepComponent} from "./to-node-step/to-node-step.component";
import {NgModule} from "@angular/core";
import {FromNodeStepComponent} from "./from-node-step/from-node-step.component";
import {PropertiesStepComponent} from "./properties-step/properties-step.component";
import {ConnectionWizardService} from "./connection-wizard.service";
import {SelectRequirementOrCapabilityModule} from "../../../../components/logic/select-requirement-or-capability/select-requirement-or-capability.module";
import {PropertyTableModule} from "../../../../components/logic/properties-table/property-table.module";
import {FormElementsModule} from "../../../../components/ui/form-components/form-elements.module";
import {ConnectionWizardHeaderComponent} from "./connection-wizard-header/connection-wizard-header.component";
import {ConnectionPropertiesViewComponent} from "./connection-properties-view/connection-properties-view.component";
import {BrowserModule} from "@angular/platform-browser";

@NgModule({
    declarations: [
        FromNodeStepComponent,
        ToNodeStepComponent,
        PropertiesStepComponent,
        ConnectionWizardHeaderComponent,
        ConnectionPropertiesViewComponent
    ],
    imports: [
        FormElementsModule,
        PropertyTableModule,
        SelectRequirementOrCapabilityModule,
        BrowserModule
    ],
    exports: [
        FromNodeStepComponent,
        ToNodeStepComponent,
        PropertiesStepComponent,
        ConnectionWizardHeaderComponent,
        ConnectionPropertiesViewComponent
    ],
    entryComponents: [FromNodeStepComponent,
        ToNodeStepComponent,
        PropertiesStepComponent,
        ConnectionWizardHeaderComponent,
        ConnectionPropertiesViewComponent
    ],
    providers: [ConnectionWizardService]
})
export class ConnectionWizardModule {
}