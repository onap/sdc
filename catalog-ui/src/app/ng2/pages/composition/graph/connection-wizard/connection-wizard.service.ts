import * as _ from "lodash";
import {ConnectRelationModel} from "app/models/graph/connectRelationModel";
import {Injectable} from "@angular/core";
import { Requirement, Capability} from "app/models";
import {Dictionary} from "lodash";
import {Match, Component, PropertyFEModel} from "app/models";
import {Store} from "@ngxs/store";
import {WorkspaceService} from "../../../workspace/workspace.service";

@Injectable()
export class ConnectionWizardService {
    
    connectRelationModel:ConnectRelationModel;
    selectedMatch:Match;
    changedCapabilityProperties:PropertyFEModel[];


    constructor(private workspaceService: WorkspaceService) {
        this.changedCapabilityProperties = [];

    }

    public setRelationMenuDirectiveObj = (connectRelationModel:ConnectRelationModel) => {
        this.connectRelationModel = connectRelationModel;
        // this.selectedCapability = rel
    }

    getOptionalRequirementsByInstanceUniqueId = (isFromTo: boolean, matchWith?:Capability): Dictionary<Requirement[]> => {
       let requirements: Array<Requirement> = [];
        _.forEach(this.connectRelationModel.possibleRelations, (match: Match) => {
            if(!matchWith || match.capability.uniqueId == matchWith.uniqueId){
                if(match.isFromTo == isFromTo){
                    requirements.push(match.requirement);
                }
            }
        });
        requirements = _.uniqBy(requirements, (req:Requirement)=>{
            return req.ownerId + req.uniqueId + req.name;
        });
        return _.groupBy(requirements, 'capability');
    }

    getOptionalCapabilitiesByInstanceUniqueId = (isFromTo: boolean, matchWith?:Requirement): Dictionary<Capability[]> => {
        let capabilities: Array<Capability> = [];
        _.forEach(this.connectRelationModel.possibleRelations, (match: Match) => {
            if(!matchWith || match.requirement.uniqueId == matchWith.uniqueId){
                if(match.isFromTo == isFromTo){
                    capabilities.push(match.capability);
                }
            }
        });
        capabilities = _.uniqBy(capabilities, (cap:Capability)=>{
            return cap.ownerId + cap.uniqueId + cap.name;
        });
        return _.groupBy(capabilities, 'type');
    }
}

