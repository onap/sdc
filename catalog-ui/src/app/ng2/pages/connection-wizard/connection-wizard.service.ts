import * as _ from "lodash";
import {ConnectRelationModel} from "../../../models/graph/connectRelationModel";
import {Injectable} from "@angular/core";
import { Requirement, Capability} from "app/models";
import {Dictionary} from "lodash";
import {Match, Component, PropertyFEModel} from "app/models";

@Injectable()
export class ConnectionWizardService {
    
    connectRelationModel:ConnectRelationModel;
    currentComponent:Component;
    selectedMatch:Match;
    changedCapabilityProperties:PropertyFEModel[];

    constructor() {
        this.changedCapabilityProperties = [];
    }

    public setRelationMenuDirectiveObj = (connectRelationModel:ConnectRelationModel) => {
        this.connectRelationModel = connectRelationModel;
        // this.selectedCapability = rel
    }


    // getComponentInstanceIdOfSelectedCapability = (): string => {
    //     if(this.selectedMatch.capability){
    //         if(this.selectedMatch.isFromTo) {
    //             return this.selectedMatch.toNode;
    //         } else {
    //             return this.selectedMatch.fromNode;
    //         }
    //     }
    //     return '';
    //
    // }

    getOptionalRequirementsByInstanceUniqueId = (isFromTo: boolean, matchWith?:Capability): Dictionary<Requirement[]> => {
       let requirements: Array<Requirement> = [];
        _.forEach(this.connectRelationModel.possibleRelations, (match: Match) => {
            if(!matchWith || match.capability.uniqueId == matchWith.uniqueId){
                if(match.isFromTo == isFromTo){
                    requirements.push(match.requirement);
                }
            }
        });
        requirements = _.uniqBy(requirements, (req:Requirement)=>{
            return req.ownerId + req.uniqueId + req.name;
        });
        return _.groupBy(requirements, 'capability');
    }

    getOptionalCapabilitiesByInstanceUniqueId = (isFromTo: boolean, matchWith?:Requirement): Dictionary<Capability[]> => {
        let capabilities: Array<Capability> = [];
        _.forEach(this.connectRelationModel.possibleRelations, (match: Match) => {
            if(!matchWith || match.requirement.uniqueId == matchWith.uniqueId){
                if(match.isFromTo == isFromTo){
                    capabilities.push(match.capability);
                }
            }
        });
        capabilities = _.uniqBy(capabilities, (cap:Capability)=>{
            return cap.ownerId + cap.uniqueId + cap.name;
        });
        return _.groupBy(capabilities, 'type');
    }
}

