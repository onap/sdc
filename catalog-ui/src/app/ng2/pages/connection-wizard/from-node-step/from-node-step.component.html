<select-requirement-or-capability [optionalRequirementsMap]="optionalRequirementsMap"
                                  [optionalCapabilitiesMap]="optionalCapabilitiesMap"
                                  [selectedReqOrCapModel]="connectWizardService.selectedMatch && (connectWizardService.selectedMatch.isFromTo ? connectWizardService.selectedMatch.requirement : connectWizardService.selectedMatch.capability)"
                                  [currentComponent]="connectWizardService.currentComponent"
                                  [componentInstanceId]="connectWizardService.connectRelationModel.fromNode.componentInstance.uniqueId"
                                  (updateSelectedReqOrCap)="updateSelectedReqOrCap($event)"
                                  (updateCapabilityProperties)="onCapabilityPropertiesUpdate($event)">
</select-requirement-or-capability>