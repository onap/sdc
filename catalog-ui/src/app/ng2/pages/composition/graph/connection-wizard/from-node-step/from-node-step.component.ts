import { Component, forwardRef, Inject, OnInit } from '@angular/core';
import { Match } from 'app/models';
import { Capability } from 'app/models/capability';
import { Requirement } from 'app/models/requirement';
import { IStepComponent } from 'app/models/wizard-step';
import { Dictionary } from 'lodash';
import { ConnectionWizardService } from '../connection-wizard.service';

@Component({
    selector: 'from-node-step',
    templateUrl: './from-node-step.component.html'
})

export class FromNodeStepComponent implements IStepComponent, OnInit{

    optionalRequirementsMap: Dictionary<Requirement[]>;
    optionalCapabilitiesMap: Dictionary<Capability[]>;

    constructor(@Inject(forwardRef(() => ConnectionWizardService)) public connectWizardService: ConnectionWizardService) {}

    ngOnInit() {
        this.optionalRequirementsMap = this.connectWizardService.getOptionalRequirementsByInstanceUniqueId(true);
        this.optionalCapabilitiesMap = this.connectWizardService.getOptionalCapabilitiesByInstanceUniqueId(false);
    }

    preventNext = (): boolean => {
        return !this.connectWizardService.selectedMatch || (!this.connectWizardService.selectedMatch.capability && !this.connectWizardService.selectedMatch.requirement);
    }

    preventBack = (): boolean => {
        return true;
    }

    private updateSelectedReqOrCap = (selected: Requirement|Capability): void => {
        if (!selected) {
            this.connectWizardService.selectedMatch = null;
        } else if (selected instanceof Requirement) {
            this.connectWizardService.selectedMatch = new Match(<Requirement>selected, null, true, this.connectWizardService.connectRelationModel.fromNode.componentInstance.uniqueId, null);
        } else {
            this.connectWizardService.selectedMatch = new Match(null, <Capability>selected , false, null, this.connectWizardService.connectRelationModel.fromNode.componentInstance.uniqueId);
        }
    }

}
