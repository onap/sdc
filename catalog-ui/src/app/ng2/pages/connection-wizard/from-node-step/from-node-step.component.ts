import {Component, OnInit, Inject, forwardRef} from "@angular/core";
import {IStepComponent} from "../../../../models/wizard-step";
import {Dictionary} from "lodash";
import { Match} from "app/models";
import {ConnectionWizardService} from "../connection-wizard.service";
import {Requirement} from "../../../../models/requirement";
import {Capability} from "../../../../models/capability";
import {PropertyModel} from "../../../../models/properties";

@Component({
    selector: 'from-node-step',
    templateUrl: './from-node-step.component.html'
})

export class FromNodeStepComponent implements IStepComponent, OnInit{

    constructor(@Inject(forwardRef(() => ConnectionWizardService)) public connectWizardService: ConnectionWizardService) {}

    optionalRequirementsMap: Dictionary<Requirement[]>;
    optionalCapabilitiesMap: Dictionary<Capability[]>;

    ngOnInit(){
        this.optionalRequirementsMap = this.connectWizardService.getOptionalRequirementsByInstanceUniqueId(true);
        this.optionalCapabilitiesMap = this.connectWizardService.getOptionalCapabilitiesByInstanceUniqueId(false);
    }

    preventNext = ():boolean => {
        return !this.connectWizardService.selectedMatch || (!this.connectWizardService.selectedMatch.capability && !this.connectWizardService.selectedMatch.requirement);
    }

    preventBack = ():boolean => {
        return true;
    }

    onCapabilityPropertiesUpdate(capabilityProperties:Array<PropertyModel>) {
        this.connectWizardService.selectedMatch.capabilityProperties = capabilityProperties;
    }

    private updateSelectedReqOrCap = (selected:Requirement|Capability):void => {
        if(!selected){
            this.connectWizardService.selectedMatch = null;
        } else if(selected instanceof Requirement){
            this.connectWizardService.selectedMatch = new Match(<Requirement>selected, null, true, this.connectWizardService.connectRelationModel.fromNode.componentInstance.uniqueId, null);
        } else{
            this.connectWizardService.selectedMatch = new Match(null,<Capability>selected , false, null, this.connectWizardService.connectRelationModel.fromNode.componentInstance.uniqueId);
        }
    }

}
