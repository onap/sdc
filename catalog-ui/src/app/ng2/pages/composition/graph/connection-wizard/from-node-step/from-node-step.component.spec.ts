import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { Capability, Match } from 'app/models';
import { ConfigureFn, configureTests } from '../../../../../../../jest/test-config.helper';
import { Requirement } from '../../../../../../models/requirement';
import { ConnectionWizardService } from '../connection-wizard.service';
import { FromNodeStepComponent } from './from-node-step.component';

describe('from-node-step component', () => {

    let fixture: ComponentFixture<FromNodeStepComponent>;
    let connectionWizardServiceMockWithoutSelectedMatch: Partial<ConnectionWizardService>;
    let connectionWizardServiceMockWithSelectedMatch: Partial<ConnectionWizardService>;

    const connectionWizardServiceMockSelectedMatchWithRequirements = {requirement: 'val'};

    connectionWizardServiceMockWithoutSelectedMatch = {
        getOptionalRequirementsByInstanceUniqueId: jest.fn().mockReturnValue(5),
        getOptionalCapabilitiesByInstanceUniqueId: jest.fn().mockReturnValue(10),

        connectRelationModel: {
            fromNode: {
                componentInstance: {
                    uniqueId : 'testUniqueID'
                }
            }
        }
    };

    connectionWizardServiceMockWithSelectedMatch = {
        selectedMatch: connectionWizardServiceMockSelectedMatchWithRequirements,
        getOptionalRequirementsByInstanceUniqueId: jest.fn().mockReturnValue(5),
        getOptionalCapabilitiesByInstanceUniqueId: jest.fn().mockReturnValue(10)
    };

    let expectedConnectionWizardServiceMock = connectionWizardServiceMockWithoutSelectedMatch;

    beforeEach(
        async(() => {
            const configure: ConfigureFn = testBed => {
                testBed.configureTestingModule({
                    declarations: [FromNodeStepComponent],
                    imports: [],
                    schemas: [NO_ERRORS_SCHEMA],
                    providers: [
                        {provide: ConnectionWizardService, useValue: expectedConnectionWizardServiceMock}
                    ],
                });
            };

            configureTests(configure).then(testBed => {
                fixture = testBed.createComponent(FromNodeStepComponent);
            });
        })
    );


    it('should match current snapshot', () => {
        expect(fixture).toMatchSnapshot();
    });

    it('preventBack return true - always', () => {
        fixture.componentInstance.ngOnInit();
        const result = fixture.componentInstance.preventBack();
        expect(result).toEqual(true);
    });

    it('preventNext return true since selectedMatch does not exist in connectionWizardServiceMock', () => {
        fixture.componentInstance.ngOnInit();
        const result = fixture.componentInstance.preventNext();
        expect(result).toEqual(true);
    });

    it('preventNext return false since to selectedMatch or selectedMatch.capability & selectedMatch.requirement does exist in connectionWizardServiceMock', () => {
        fixture.componentInstance.connectWizardService = connectionWizardServiceMockWithSelectedMatch;
        fixture.componentInstance.ngOnInit();
        const result = fixture.componentInstance.preventNext();
        expect(result).toEqual(false);
    });

    it('updateSelectedReqOrCap is called with instance of requirement, the selectMatch will be set to an Instance of Match of type Requirement', () => {
        const requirement = new Requirement();
        fixture.componentInstance.updateSelectedReqOrCap(requirement);
        const expectedSelectedMatch = fixture.componentInstance.connectWizardService.selectedMatch;

        expect(expectedSelectedMatch).toBeInstanceOf(Match);
        expect(expectedSelectedMatch.capability).toBe(null);
        expect(expectedSelectedMatch.fromNode).toBe('testUniqueID');
        expect(expectedSelectedMatch.isFromTo).toBe(true);
        expect(expectedSelectedMatch.toNode).toBe(null);
        expect(expectedSelectedMatch.requirement).toBeInstanceOf(Requirement);
    });

    it('updateSelectedReqOrCap is called with instance of capability, the selectMatch will be set to an Instance of Match of type Capability', () => {
        const capability = new Capability();
        fixture.componentInstance.updateSelectedReqOrCap(capability);
        const expectedSelectedMatch = fixture.componentInstance.connectWizardService.selectedMatch;

        expect(expectedSelectedMatch).toBeInstanceOf(Match);
        expect(expectedSelectedMatch.requirement).toBe(null);
        expect(expectedSelectedMatch.fromNode).toBe(null);
        expect(expectedSelectedMatch.isFromTo).toBe(false);
        expect(expectedSelectedMatch.toNode).toBe('testUniqueID');
        expect(expectedSelectedMatch.capability).toBeInstanceOf(Capability);
    });

    it('updateSelectedReqOrCap is called with null, the selectMatch will be set to null', () => {
        fixture.componentInstance.updateSelectedReqOrCap(null);
        const expectedSelectedMatch = fixture.componentInstance.connectWizardService.selectedMatch;

        expect(expectedSelectedMatch).toBe(null);
    });

});