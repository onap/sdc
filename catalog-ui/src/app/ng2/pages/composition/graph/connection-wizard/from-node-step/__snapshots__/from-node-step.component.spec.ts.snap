// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`from-node-step component should match current snapshot 1`] = `
<from-node-step
  connectWizardService={[Function Object]}
  preventBack={[Function Function]}
  preventNext={[Function Function]}
  updateSelectedReqOrCap={[Function Function]}
>
  <select-requirement-or-capability />
</from-node-step>
`;
